packages/ses/src/console.js

```diff
@@ -0,0 +1,102 @@
+import { globalThis, TypeError } from './commons.js';
+
+/**
+ * @import {InternalConsole, ExternalConsole} from './console-types.js'
+ */
+
+/**
+ * Creates a suitable console for internal errors and warnings out of the
+ * Node.js console.error to ensure all messages to go stderr, including the
+ * group label.
+ * Accounts for the extra space introduced by console.error as a delimiter
+ * between the indent and subsequent arguments.
+ *
+ * @param {(...message: Array<any>) => void} log logs to stderr
+ */
+const adaptConsole = log => {
+  let indent = false;
+  /** @param {Array<any>} args */
+  const indentLog = (...args) => {
+    if (indent) {
+      log(' ', ...args);
+    } else {
+      log(...args);
+    }
+  };
+  return /** @type {ExternalConsole} */ ({
+    warn(...args) {
+      indentLog(...args);
+    },
+    error(...args) {
+      indentLog(...args);
+    },
+    groupCollapsed(...args) {
+      indentLog(...args);
+      indent = true;
+    },
+    groupEnd() {
+      indent = false;
+    },
```

Comment:
Given that this console is for internal use and we only ever indent once, I’m inclined to keep this simple and add an error check. I left the string manipulation on the shop floor since it’s weird and unnecessary. But, it looks like accumulating two spaces for each level, and always slicing one off for an indented log since the Node.js console itself always adds one space between arguments. In my opinion, it’s too weird to implement given that it’ll never be used.